File: tests/llvm/hardlift/benchmark/numba/color_burn_8.py
from numba import njit

@njit
def color_burn_8(base, active):
  output = []
  m = len(base)
  n = len(base[0])
  for i in range(m):
    curr_row = []
    for j in range(n):
      if active[i][j] == 0:
        pixel = 255
      else:
        pixel = 255 - (255 - base[i][j]) / active[i][j]
      curr_row.append(pixel)
    output.append(curr_row)
  return output

File: tenspiler/llm/benchmarks/blend/source/color_burn_8.cc
#include <vector>
using namespace std;

vector<vector<int>> color_burn_8(vector<vector<int>> base, vector<vector<int>> active)
{
    vector<vector<int>> out;
    int m = base.size();
    int n = base[0].size();
	for (int row = 0; row < m; row++) {
        vector<int> row_vec;
		for (int col = 0; col < n; col++) {
            int pixel;
            if (active[row][col] == 0)
                pixel = 32;
            else
                pixel = 32 - (32 - base[row][col]) / active[row][col];
            row_vec.push_back(pixel);
		}
		out.push_back(row_vec);
	}
	return out;
}
